executor/new_executor.go

```diff
@@ -112,7 +133,55 @@ func (e *HashJoinExec) Fields() []*ast.ResultField {
 	return nil
 }
 
+var batchSize = 100
+
+// Worker to get big table rows.
+func (e *HashJoinExec) fetchBigExec() {
+	cnt := 0
+	defer func() {
+		for _, cn := range e.bigTableRows {
+			close(cn)
+		}
+	}()
+	for {
+		if e.finished {
+			break
+		}
+		rows := make([]*Row, 0, batchSize)
+		done := false
+		for i := 0; i < batchSize; i++ {
```

Comment:
It is better to use a slow start algorithm for batchSize. For example the batchSize is 1, 2, 4, ...., maxBatchSize for each loop. The join worker will get data earlier. @hanfei1991